pieces/shared/menu-button/trigger.tsx

```diff
@@ -0,0 +1,88 @@
+import {
+  KeyboardEventHandler,
+  MouseEventHandler,
+  RefObject,
+  useCallback,
+  useEffect,
+  useRef,
+} from 'react';
+
+import Button from '../button';
+import moreIcon from '../more.svg';
+
+type TriggerProps = {
+  isOpen: boolean;
+  label: string;
+  openMenu: ({ initialFocus }: { initialFocus: 'first' | 'last' }) => void;
+  shouldResetFocus: boolean;
+};
+
+/**
+ * Handles moving focus back to the trigger on menu close.
+ */
+function useFocusOnClose(
+  ref: RefObject<HTMLUnknownElement>,
+  {
+    isOpen,
+    shouldResetFocus,
+  }: {
+    isOpen: boolean;
+    shouldResetFocus: boolean;
+  },
+) {
+  useEffect(() => {
+    if (!isOpen && shouldResetFocus) {
+      // Running in an 'immediate' timeout so that
+      // when the menu has closed because of an enter key press,
+      // it won't trigger a click event on the trigger as well.
+      setTimeout(() => {
```

Comment:
I wonder if there is a way to do this without relying on a setTimeout. All good to leave this though

Reply:
I think `event.preventDefault()` on the `keydown` event works as well.